BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/location/Location.java

```diff
@@ -55,13 +61,52 @@ public Location(Location location) {
      * @return distance in meters
      */
     public double getDistanceTo(Location location) {
-        return LocationDistanceCalculator.calculateDistanceBetween(this, location, false);
+        return LocationDistanceCalculator.calculateDistanceBetween(this, location);
     }
 
     public double getAngleTo(Location location) {
         return getRotationAngleInDegrees(this, location);
     }
 
+    /**
+     * Shifts the current {@link #latitude} and {@link #longitude} based on the specified distance
+     * and angle.
+     *
+     * @param distance in meters
+     * @param angle    in degrees [0°-360°)
+     * @see <a href="http://mathworld.wolfram.com/GreatCircle.html">Great Circel Wolfram Alpha</a>
+     * @see <a href="https://en.wikipedia.org/wiki/Great-circle_navigation#Finding_way-points">Great
+     * Circle Wikipedia</a>
+     */
+    public void shift(double distance, double angle) {
+        angle = angle % 360;
+        double bearingRadians = Math.toRadians(angle);
+        double latitudeRadians = Math.toRadians(latitude);
+        double longitudeRadians = Math.toRadians(longitude);
+        // convert distance to km and calculate fraction of earth radius
+        double distanceFraction = (distance / 1000) / LocationDistanceCalculator.EARTH_RADIUS;
+        double shiftedLatitudeRadians = Math.asin(Math.sin(latitudeRadians) * Math.cos(distanceFraction) +
+                Math.cos(latitudeRadians) * Math.sin(distanceFraction) * Math.cos(bearingRadians));
+        double shiftedLongitudeRadians = longitudeRadians + Math.atan2(Math.sin(bearingRadians) * Math.sin(distanceFraction) *
```

Comment:
According to the calculation [here](http://lists.maptools.org/pipermail/proj/2008-October/003939.html) it is `λ₂ = λ₁ + atan2 [sin c sin Az, (cos φ₁ cos c - sin φ₁ sin c cos Az)]`. The equivalent should be `longitudeRadians + Math.atan2(Math.sin(distanceFraction) * Math.sin(bearingRadians), Math.cos(latitudeRadians), Math.cos(distanceFraction) - Math.sin(latitudeRadians) * Math.sin(distanceFraction) * Math.cos(bearingRadians));`.